src/tests/helper/net.rs

```diff
@@ -0,0 +1,76 @@
+use crate::net::{ConnectionManager, ConnectionManagerError, Message, SignerID};
+use redis::ControlFlow;
+use std::cell::RefCell;
+use std::sync::mpsc::Receiver;
+use std::thread;
+use std::thread::JoinHandle;
+
+pub struct TestConnectionManager {
+    should_broadcast: Vec<Message>,
+    pub broadcasted: RefCell<Vec<Message>>,
+
+    should_send: Vec<Message>,
+    pub sent: RefCell<Vec<Message>>,
+}
+
+impl TestConnectionManager {
+    pub fn new() -> Self {
+        Self {
+            should_broadcast: vec![],
+            broadcasted: RefCell::new(vec![]),
+
+            should_send: vec![],
+            sent: RefCell::new(vec![]),
+        }
+    }
+
+    pub fn assert(self) {
+        if let TestConnectionManager {
```

Comment:
Executing `cargo test` reports three warnings .

```
warning: irrefutable if-let pattern
  --> src/tests/helper/net.rs:28:9
   |
28 | /         if let TestConnectionManager {
29 | |             should_broadcast,
30 | |             broadcasted,
31 | |             should_send,
...  |
36 | |             assert_eq!(should_send, sent.into_inner());
37 | |         }
   | |_________^
   |
   = note: `#[warn(irrefutable_let_patterns)]` on by default

warning: unused variable: `message_processor`
  --> src/tests/helper/net.rs:64:13
   |
64 |         mut message_processor: impl FnMut(Message) -> ControlFlow<()> + Send + 'static,
   |             ^^^^^^^^^^^^^^^^^ help: consider prefixing with an underscore: `_message_processor`
   |
   = note: `#[warn(unused_variables)]` on by default

warning: variable does not need to be mutable
  --> src/tests/helper/net.rs:64:9
   |
64 |         mut message_processor: impl FnMut(Message) -> ControlFlow<()> + Send + 'static,
   |         ----^^^^^^^^^^^^^^^^^
   |         |
   |         help: remove this `mut`
   |
   = note: `#[warn(unused_mut)]` on by default
```

Reply:
It was fixed!